dkron/store.go

```diff
@@ -125,6 +125,9 @@ func (s *Store) SetJob(job *Job, copyDependentJobs bool) error {
 	}
 
 	err := s.db.Update(func(tx *buntdb.Tx) error {
+		// set pristine as default for newly created jobs
+		job.Status = "pristine"
```

Comment:
Yes, the job status gets updated according to execution status, but what is the problem with the default value getting lost? AFAIK the default value only gets used when a new job was created.

And secondly, I did it this way to allow by default **all** jobs (success/failed/untriggered) to continue showing without specifying filters.

Reply:
The scenario I mean is when there is a job that is untriggered, and then dkron gets updated to the version with this change. That same untriggered job will still have status "" instead of the expected "pristine"/"untriggered".
I've tried to create this situation using this change and I found that the replay of the raft log is making it hard to reproduce. I need a bit more time to get a working example to prove (or disprove :) my point.

Reply:
Here's the problem:
4 jobs, 1 successful, 1 failed, 1 untriggered (created before this change), 1 pristine (created after this change):
![image](https://user-images.githubusercontent.com/4980327/111083654-e43a6280-850e-11eb-85b9-45f7d90b666a.png)

Filter these on Waiting to Run:
![image](https://user-images.githubusercontent.com/4980327/111083718-2e234880-850f-11eb-95cc-8b3b553d6aa2.png)

Only the one created after the change shows up, because the other one still has a Status of "".

This will occur when an untriggered job exists AND was persisted in a snapshot by raft before updating.

Reply:
Ok, I will do the required changes